synapse/storage/databases/main/deviceinbox.py

```diff
@@ -290,7 +303,7 @@ def get_device_messages_txn(txn: LoggingTransaction):
             # If a list of device IDs was not provided, retrieve all devices IDs
             # for the given users. We explicitly do not query hidden devices, as
             # hidden devices should not receive to-device messages.
```

Comment:
```suggestion
            # hidden devices should not receive to-device messages.
            # Note that this is more efficient than just dropping `device_id` from the query,
            # since device_inbox has an index on `(user_id, device_id, stream_id)`
```